apps/sample-app/src/components/Emails.tsx

```diff
@@ -0,0 +1,49 @@
+import { DocumentCard, DocumentCardTitle } from '@fluentui/react';
+import { Text } from '@fluentui/react-components';
+import { Message } from '@microsoft/microsoft-graph-types';
+import React from 'react';
+
+import { handleOpenMailItem } from './utils';
+
+interface EmailProps {
+  messages: Message[];
+}
+export interface MessageListItem {
+  key?: string;
+  subject?: string;
+  sender?: string;
+}
+export const EmailList: React.FC<EmailProps> = (props: EmailProps) => {
+  const { messages } = props;
+  const emailItems = messages.map<MessageListItem>(m => {
+    return {
+      key: m.id,
+      subject: m.subject || '',
+      sender: m.sender?.emailAddress?.name || '',
+    };
```

Comment:
If you're interested in being super-efficient without (I don't think) sacrificing readability here you can actually get rid of the return statement and just replace it with parentheses:
```
  const emailItems = messages.map<MessageListItem>(m => ({
      key: m.id,
      subject: m.subject || '',
      sender: m.sender?.emailAddress?.name || '',
    }));
```